tests/system/providers/amazon/aws/example_bedrock_knowledge_base.py

```diff
@@ -0,0 +1,519 @@
+    """
+    log.info("Deleting Knowledge Base %s.", knowledge_base_id)
+    bedrock_agent_client.delete_knowledge_base(knowledgeBaseId=knowledge_base_id)
+
+
+# [END howto_operator_bedrock_delete_knowledge_base]
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def delete_vector_index(collection_id: str):
+    """
+    Delete the vector index created earlier.
+
+    :param collection_id: ID of the collection to be indexed.
+    """
+    host = f"{collection_id}.{region_name}.aoss.amazonaws.com"
+    credentials = boto3.Session().get_credentials()
+    awsauth = AWSV4SignerAuth(credentials, region_name, "aoss")
+
+    # Build the OpenSearch client
+    oss_client = OpenSearch(
+        hosts=[{"host": host, "port": 443}],
+        http_auth=awsauth,
+        use_ssl=True,
+        verify_certs=True,
+        connection_class=RequestsHttpConnection,
+        timeout=300,
+    )
+    oss_client.indices.delete(index=index_name)
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def delete_collection(collection_id: str):
+    """
+    Delete the OpenSearch collection created earlier.
+
+    :param collection_id: ID of the collection to be indexed.
+    """
+    log.info("Deleting collection %s.", collection_id)
+    aoss_client.delete_collection(id=collection_id)
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def delete_opensearch_policies():
+    """Delete the security, network and data access policies created earlier."""
+
+    access_policies = aoss_client.list_access_policies(
+        type="data", resource=[f"collection/{vector_store_name}"]
+    )["accessPolicySummaries"]
+    log.info("Found access policies for %s: %s", vector_store_name, access_policies)
+    if not access_policies:
+        raise Exception("No access policies found?")
+    for policy in access_policies:
+        log.info("Deleting access policy for %s: %s", vector_store_name, policy["name"])
+        aoss_client.delete_access_policy(name=policy["name"], type="data")
+
+    for policy_type in ["encryption", "network"]:
+        policies = aoss_client.list_security_policies(
+            type=policy_type, resource=[f"collection/{vector_store_name}"]
+        )["securityPolicySummaries"]
+        if not policies:
+            raise Exception("No security policies found?")
+        log.info("Found %s security policies for %s: %s", policy_type, vector_store_name, policies)
+        for policy in policies:
+            log.info("Deleting %s security policy for %s: %s", policy_type, vector_store_name, policy["name"])
+            aoss_client.delete_security_policy(name=policy["name"], type=policy_type)
+
+
+with DAG(
+    dag_id=DAG_ID,
+    schedule="@once",
+    start_date=datetime(2021, 1, 1),
+    tags=["example"],
+    catchup=False,
+) as dag:
+    test_context = sys_test_context_task()
+    env_id = test_context["ENV_ID"]
+
+    aoss_client = OpenSearchServerlessHook().conn
+    bedrock_agent_client = BedrockAgentHook().conn
```

Comment:
This two lines responses for calls to the DB. If you would like to avoid it you need to move it inside of the tasks, where it required.

Or if you intends to use only default `botocore` strategy you might explicitly specify it by provide `aws_conn_id=None`

Reply:
Looks like that did it.  Cheers.

Reply:
Yeah, `aws_con_id=None` skip get connection from the DB / env variable / secrets backends and only use configuration from hook